---
title: "Analysis for paper"
author: "Lizzie Coker"
date: "20/04/2022"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(plyr)
library(reshape2)
library(scales)
library(dplyr)
library(ggplot2)
library(superheat)
library(rowr)
library(ggplot2)
library(dplyr)
library(magrittr)
library(knitr)
library(lubridate)
library(tidyr)
library(RColorBrewer)
library(reshape2)
library(knitr)
library(kableExtra)
library(scales)
library(ggpubr)
library(stats)
library(ggridges)
library(hexbin)
set.seed(13)
```

## Data upload
```{r}
sand1<-read.csv("Input/sandpiper-01_matrix_results.csv")
sand1$screen<-"Sandpiper_1"
g7<-read.csv("Input/gdsc-007_matrix_results.csv")
g7$screen<-"GDSC007"
g8<-read.csv("Input/gdsc-008_matrix_results.csv")
g8$screen<-"GDSC008"
g9<-read.csv("Input/gdsc-009_matrix_results.csv")
g9$screen<-"GDSC009"
g10<-read.csv("Input/gdsc-010_matrix_results.csv")
g10$screen<-"GDSC010"

data<-rbind(sand1, g7, g8, g9, g10)
#write.csv(data, "Supplementary Table 3 - single and combination metrics.csv")
```

## Overall stats
```{r}
#Number of cell lines
length(unique(data$CELL_LINE_NAME))
#Number of cancer types
length(unique(data$CANCER_TYPE))
#Number of combinations
data$combo_id<-paste(data$lib1_name, data$lib2_name, sep=" + ")
length(unique(data$combo_id))
#Number of combinations-cell line pairs
data$combo_id_cl<-paste(data$combo_id, data$CELL_LINE_NAME, sep=", ")
length(unique(data$combo_id_cl))
#Numbers of combinations and cell lines in the full and half panels
full_panel<-subset(data, data$screen!="GDSC010")
length(unique(full_panel$CELL_LINE_NAME))
length(unique(full_panel$combo_id))
half_panel<-subset(data, data$screen=="GDSC010")
length(unique(half_panel$CELL_LINE_NAME))
length(unique(half_panel$combo_id))

#Counting number of instances where Bliss window is positive and Bliss matrix is negative
nrow(data[(data$Bliss_window > 0 & data$Bliss_matrix<0),])
43967/84055


#Making table of cell lines
cell_lines<-select(data, CELL_LINE_NAME, COSMIC_ID, SIDM, TISSUE, CANCER_TYPE)
cell_lines<-distinct(cell_lines)
write.csv(cell_lines, "Cell_lines_used.csv", row.names = FALSE)

#Making table of combinations
combo_list<-select(data, lib1_ID, lib1_name, lib1_owner, lib2_ID, lib2_name, lib2_owner, combo_id, screen)
combo_list<-distinct(combo_list)
write.csv(combo_list, "Combinations_used.csv", row.names = FALSE)

```
## Fig 1B - note additional annotation in Adobe Illustrator
```{r}
#Making frequency table of cell line cancer types
cell_lines_freq<-cell_lines %>%
  group_by(CANCER_TYPE) %>%
  summarize(Freq=n()) %>%
  arrange(desc(Freq))


ggplot(cell_lines_freq, aes(x="", y=Freq, fill=reorder(CANCER_TYPE, Freq))) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=0) +
  theme_void() + 
  theme(legend.position="none") 

```

## Making matrices of values - example input file for GDSCTools
```{r}
#For each metric used for biomarker input, need to create matrices of these inputs for GDSCTools ANOVAs
sand1_combo_Emax_matrix<-sand1
sand1_combo_Emax_matrix$Combination<-paste(sand1_combo_Emax_matrix$lib1_ID, sand1_combo_Emax_matrix$lib2_ID, sep="")
sand1_combo_Emax_matrix<-select(sand1_combo_Emax_matrix, COSMIC_ID, Combination, combo_MaxE)
sand1_combo_Emax_matrix<-aggregate(.~COSMIC_ID + Combination, sand1_combo_Emax_matrix, mean)
sand1_combo_Emax_matrix$COSMIC_ID<-paste0('abc', sand1_combo_Emax_matrix$COSMIC_ID)
an <- with(sand1_combo_Emax_matrix, sort(unique(c(as.character(COSMIC_ID), as.character(Combination)))))
M <- array(0, c(length(an), length(an)), list(an, an))
i <- match(sand1_combo_Emax_matrix$COSMIC_ID, an)
j <- match(sand1_combo_Emax_matrix$Combination, an)
M[cbind(i,j)] <- M[cbind(j,i)] <- sand1_combo_Emax_matrix$combo_MaxE
newM<-M[, -grep("abc", colnames(M))]
thirdM<-newM[grep("abc", row.names(newM)),]
row.names(thirdM)<-sub("abc", "", row.names(thirdM))
write.csv(thirdM, "Biomarker_inputs/Sandpiper1_Combo_MaxE_matrix.csv")


g07_combo_Emax_matrix<-g7
g07_combo_Emax_matrix$Combination<-paste(g07_combo_Emax_matrix$lib1_ID, g07_combo_Emax_matrix$lib2_ID, sep="")
g07_combo_Emax_matrix<-select(g07_combo_Emax_matrix, COSMIC_ID, Combination, combo_MaxE)
g07_combo_Emax_matrix<-aggregate(.~COSMIC_ID + Combination, g07_combo_Emax_matrix, mean)
g07_combo_Emax_matrix$COSMIC_ID<-paste0('abc', g07_combo_Emax_matrix$COSMIC_ID)
an <- with(g07_combo_Emax_matrix, sort(unique(c(as.character(COSMIC_ID), as.character(Combination)))))
M <- array(0, c(length(an), length(an)), list(an, an))
i <- match(g07_combo_Emax_matrix$COSMIC_ID, an)
j <- match(g07_combo_Emax_matrix$Combination, an)
M[cbind(i,j)] <- M[cbind(j,i)] <- g07_combo_Emax_matrix$combo_MaxE
newM<-M[, -grep("abc", colnames(M))]
thirdM<-newM[grep("abc", row.names(newM)),]
row.names(thirdM)<-sub("abc", "", row.names(thirdM))
write.csv(thirdM, "Biomarker_inputs/GDSC007_Combo_MaxE_matrix.csv")

g08_combo_Emax_matrix<-g8
g08_combo_Emax_matrix$Combination<-paste(g08_combo_Emax_matrix$lib1_ID, g08_combo_Emax_matrix$lib2_ID, sep="")
g08_combo_Emax_matrix<-select(g08_combo_Emax_matrix, COSMIC_ID, Combination, combo_MaxE)
g08_combo_Emax_matrix<-aggregate(.~COSMIC_ID + Combination, g08_combo_Emax_matrix, mean)
g08_combo_Emax_matrix$COSMIC_ID<-paste0('abc', g08_combo_Emax_matrix$COSMIC_ID)
an <- with(g08_combo_Emax_matrix, sort(unique(c(as.character(COSMIC_ID), as.character(Combination)))))
M <- array(0, c(length(an), length(an)), list(an, an))
i <- match(g08_combo_Emax_matrix$COSMIC_ID, an)
j <- match(g08_combo_Emax_matrix$Combination, an)
M[cbind(i,j)] <- M[cbind(j,i)] <- g08_combo_Emax_matrix$combo_MaxE
newM<-M[, -grep("abc", colnames(M))]
thirdM<-newM[grep("abc", row.names(newM)),]
row.names(thirdM)<-sub("abc", "", row.names(thirdM))
write.csv(thirdM, "Biomarker_inputs/GDSC008_Combo_MaxE_matrix.csv")

g09_combo_Emax_matrix<-g9
g09_combo_Emax_matrix$Combination<-paste(g09_combo_Emax_matrix$lib1_ID, g09_combo_Emax_matrix$lib2_ID, sep="")
g09_combo_Emax_matrix<-select(g09_combo_Emax_matrix, COSMIC_ID, Combination, combo_MaxE)
g09_combo_Emax_matrix<-aggregate(.~COSMIC_ID + Combination, g09_combo_Emax_matrix, mean)
g09_combo_Emax_matrix$COSMIC_ID<-paste0('abc', g09_combo_Emax_matrix$COSMIC_ID)
an <- with(g09_combo_Emax_matrix, sort(unique(c(as.character(COSMIC_ID), as.character(Combination)))))
M <- array(0, c(length(an), length(an)), list(an, an))
i <- match(g09_combo_Emax_matrix$COSMIC_ID, an)
j <- match(g09_combo_Emax_matrix$Combination, an)
M[cbind(i,j)] <- M[cbind(j,i)] <- g09_combo_Emax_matrix$combo_MaxE
newM<-M[, -grep("abc", colnames(M))]
thirdM<-newM[grep("abc", row.names(newM)),]
row.names(thirdM)<-sub("abc", "", row.names(thirdM))
write.csv(thirdM, "Biomarker_inputs/GDSC009_Combo_MaxE_matrix.csv")

g10_combo_Emax_matrix<-g10
g10_combo_Emax_matrix$Combination<-paste(g10_combo_Emax_matrix$lib1_ID, g10_combo_Emax_matrix$lib2_ID, sep="")
g10_combo_Emax_matrix<-select(g10_combo_Emax_matrix, COSMIC_ID, Combination, combo_MaxE)
g10_combo_Emax_matrix<-aggregate(.~COSMIC_ID + Combination, g10_combo_Emax_matrix, mean)
g10_combo_Emax_matrix$COSMIC_ID<-paste0('abc', g10_combo_Emax_matrix$COSMIC_ID)
an <- with(g10_combo_Emax_matrix, sort(unique(c(as.character(COSMIC_ID), as.character(Combination)))))
M <- array(0, c(length(an), length(an)), list(an, an))
i <- match(g10_combo_Emax_matrix$COSMIC_ID, an)
j <- match(g10_combo_Emax_matrix$Combination, an)
M[cbind(i,j)] <- M[cbind(j,i)] <- g10_combo_Emax_matrix$combo_MaxE
newM<-M[, -grep("abc", colnames(M))]
thirdM<-newM[grep("abc", row.names(newM)),]
row.names(thirdM)<-sub("abc", "", row.names(thirdM))
write.csv(thirdM, "Biomarker_inputs/GDSC010_Combo_MaxE_matrix.csv")

#Merging all matrices together
sand1_m<-read.csv("Biomarker_inputs/Sandpiper1_Combo_MaxE_matrix.csv")
g7_m<-read.csv("Biomarker_inputs/GDSC007_Combo_MaxE_matrix.csv")
g8_m<-read.csv("Biomarker_inputs/GDSC008_Combo_MaxE_matrix.csv")
g9_m<-read.csv("Biomarker_inputs/GDSC009_Combo_MaxE_matrix.csv")
g10_m<-read.csv("Biomarker_inputs/GDSC010_Combo_MaxE_matrix.csv")
colnames(sand1_m)<-gsub("X", "", colnames(sand1_m))
colnames(g7_m)<-gsub("X", "", colnames(g7_m))
colnames(g8_m)<-gsub("X", "", colnames(g8_m))
colnames(g9_m)<-gsub("X", "", colnames(g9_m))
colnames(g10_m)<-gsub("X", "", colnames(g10_m))

#adding differential labelling to distinguish between 007 and 008 screen for repeated combination
colnames(g7_m)<-gsub("24312574", "2431257407", colnames(g7_m))
colnames(g8_m)<-gsub("24312574", "2431257408", colnames(g8_m))

overall_combo_maxE<-cbind.fill(sand1_m,g7_m,g8_m,g9_m, g10_m, fill = NA)
#Removing repeated name row
overall_combo_maxE<-overall_combo_maxE[,-4]
overall_combo_maxE<-overall_combo_maxE[,-29]
overall_combo_maxE<-overall_combo_maxE[,-57]
overall_combo_maxE<-overall_combo_maxE[,-85]
colnames(overall_combo_maxE)<-gsub("X", "", colnames(overall_combo_maxE))
colnames(overall_combo_maxE)[1]<-"COSMIC_ID"
write.csv(overall_combo_maxE, "Biomarker_inputs/Sand1-GDSC010_Combo_MaxE_matrix.csv", row.names = FALSE)
```

##Figure 1D
```{r}
heatmap<-read.csv("Biomarker_inputs/Sand1-GDSC010_Combo_MaxE_matrix.csv")
rnames <- heatmap[,1]
mat_data <- data.matrix(heatmap[,2:111])
rownames(mat_data) <- rnames

superheat(mat_data, scale = FALSE, row.dendrogram = TRUE, col.dendrogram = TRUE, title = "Combo Emax no scaling", row.title = "Cell lines", column.title = "Drug combinations", legend.text.size = 7, heat.na.col = "white", bottom.label.text.angle = 90, bottom.label.text.size = 2)

```


##Figure S1E
```{r}
data_S1E_1<-data
data_S1E_2<-data
#Rename lib1 and lib2 ln IC50s as single agent ln IC50s
names(data_S1E_1)[17]<-"Sandpiper_ln_IC50"
names(data_S1E_2)[28]<-"Sandpiper_ln_IC50"
#Create compound variable of cell line ID and drug ID for matching to monotherapy data
data_S1E_1$test<-paste(data_S1E_1$SIDM, data_S1E_1$lib1_ID, sep="_")
data_S1E_2$test<-paste(data_S1E_2$SIDM, data_S1E_2$lib2_ID, sep="_")
data_S1E_1<-select(data_S1E_1, test, Sandpiper_ln_IC50)
data_S1E_2<-select(data_S1E_2, test, Sandpiper_ln_IC50)
data_S1E<-rbind(data_S1E_1, data_S1E_2)
#Aggregate replicates by mean
data_S1E<-aggregate(x = data_S1E$Sandpiper_ln_IC50,
                by = list(data_S1E$test),              
                FUN = mean)   
names(data_S1E)[2]<-"Sandpiper_ln_IC50"
names(data_S1E)[1]<-"test"

#Read in previously published single agent ln_IC50s
gdsc2<-read.csv("Input/GDSC2_fitted_dose_response_25Feb20.csv")
gdsc2<-select(gdsc2, SANGER_MODEL_ID, DRUG_ID, LN_IC50)
names(gdsc2)[3]<-"GDSC2_ln_IC50"
#Create compound variable of cell line ID and drug ID for matching to combination single agent ln IC50s
gdsc2$test<-paste(gdsc2$SANGER_MODEL_ID, gdsc2$DRUG_ID, sep="_")
gdsc2<-gdsc2[,-1:-2]
#Aggregate replicates by mean
gdsc2<-aggregate(x = gdsc2$GDSC2_ln_IC50,
                 by = list(gdsc2$test),              
                 FUN = mean)   
names(gdsc2)[2]<-"GDSC2_ln_IC50"
names(gdsc2)[1]<-"test"

all_S1E<-merge(data_S1E, gdsc2, by="test")

#correlation between combination study ln IC50s and previously-published ln IC50s
cor(all_S1E$Sandpiper_ln_IC50, all_S1E$GDSC2_ln_IC50)

plot(all_S1E$GDSC2_ln_IC50, all_S1E$Sandpiper_ln_IC50, xlab="ln(IC50) in GDSC2", ylab="ln(IC50) for single agents in this study", pch=20)


```

##Figure S1F
```{r}
data_1_scatter<-data
data_2_scatter<-data
#Renaming lib1 and lib2 Emax as single agent Emax
names(data_1_scatter)[16]<-"Single_agent_Emax"
names(data_2_scatter)[27]<-"Single_agent_Emax"
data_1_scatter<-select(data_1_scatter, Single_agent_Emax, combo_MaxE, Bliss_matrix, HSA_matrix)
data_2_scatter<-select(data_2_scatter, Single_agent_Emax, combo_MaxE, Bliss_matrix, HSA_matrix)
data_scatter<-rbind(data_1_scatter, data_2_scatter)

plot(data_scatter$Single_agent_Emax, data_scatter$combo_MaxE, xlab="Single agent Emax", ylab="Combo Emax", pch=20)


```

##Figure 1C
```{r}
data_S1G<-select(data, lib1_ID, lib1_pathway, lib2_ID, lib2_pathway)
#Identify distinct drug combinations
data_S1G<-distinct(data_S1G)
data_S1G$LIB1<-""
data_S1G$LIB2<-""
data_S1G$lib2_pathway<-gsub("Other", "Chemotherapy", data_S1G$lib2_pathway)
data_S1G$lib1_pathway<-as.character(data_S1G$lib1_pathway)
data_S1G$lib2_pathway<-as.character(data_S1G$lib2_pathway)
for (i in 1:nrow(data_S1G)){
  #Display pathway pairings in alphabetical order
  data_S1G$LIB1[i]<-pmin(data_S1G$lib1_pathway[i], data_S1G$lib2_pathway[i])
  data_S1G$LIB2[i]<-pmax(data_S1G$lib1_pathway[i], data_S1G$lib2_pathway[i])
}


a<-table(data_S1G$LIB1, data_S1G$LIB2)
a<-as.data.frame(a)
names(a)<-c("lib1_pathway", "lib2_pathway", "Frequency")

ggplot(a, aes(lib1_pathway, lib2_pathway)) + geom_point(aes(size=Frequency))+ theme(axis.text.x = element_text(angle = 90, hjust = 1)) + labs(size = "Number of combinations", x= "Pathway 1", y="Pathway 2") +scale_size_area( max_size = 10)

#Table of number of pathway combinations
a
```

##Figure S1G
```{r}
plot(data_scatter$Bliss_matrix, data_scatter$combo_MaxE, xlab="Bliss excess", ylab="Combo Emax", pch=20)
abline(v = c(0, 0.1), col=c("gray", "red"), lty=c(1,2), lwd=c(1, 3))
```

##Figure S1H
```{r}
plot(data_scatter$HSA_matrix, data_scatter$combo_MaxE, xlab="HSA excess", ylab="Combo Emax", pch=20)
abline(v = c(0, 0.1), col=c("gray", "red"), lty=c(1,2), lwd=c(1, 3))

```

##Figure S1I
```{r}
plot(data_scatter$Bliss_matrix, data_scatter$HSA_matrix, xlab="Bliss excess", ylab="HSA excess", pch=20)
abline(h = c(0, 0.1), col=c("gray", "red"), lty=c(1,2), lwd=c(1, 3))
abline(v = c(0, 0.1), col=c("gray", "red"), lty=c(1,2), lwd=c(1, 3))
cor(data_scatter$Bliss_matrix, data_scatter$HSA_matrix)
```

##Figure S1J
```{r}
plot(data_scatter$Single_agent_Emax, data_scatter$Bliss_matrix, xlab="Single agent Emax", ylab="Bliss excess", pch=20)
abline(h = c(0, 0.1), col=c("gray", "red"), lty=c(1,2), lwd=c(1, 3))
```

##Figure S1K
```{r}
a<-read.csv("Input/plate_qc_Sandpiper-01_13May22.csv")
b<-read.csv("Input/plate_qc_Sandpiper-02_13May22.csv")
data_S1L<-rbind(a,b)
data_S1L<-subset(data_S1L, data_S1L$QC_FLAG=="P")
data_S1L<-subset(data_S1L, data_S1L$RATIO_NC1_TO_PC1>=4)
data_S1L<-subset(data_S1L, data_S1L$RATIO_NC1_TO_PC2>=4)
data_S1L$label<-"DMSO, NC1"

#CV plot - remember to change dashed line according to threshold
p <- ggplot(data_S1L, aes(x=CV_NC1)) + geom_boxplot() + scale_color_grey() + theme_classic() + coord_flip() + labs(x="Coefficient of variation (CV)", ylab="DMSO\n NC1") + theme(axis.ticks = element_blank(), axis.text.x=element_blank()) + geom_vline(xintercept=0.18, linetype="dashed",  color = "gray", size=2) + xlim(0,0.2)
p

median(data_S1L$CV_NC1)
range(data_S1L$CV_NC1)
```
##Figure S1L
```{r}
data_S1L_1<-data_S1L$Z_NC1_PC1
data_S1L_2<-data_S1L$Z_NC1_PC2
data_S1L_B<-data_S1L$Z_NC1_B

data_S1L_1<-as.data.frame(data_S1L_1)
data_S1L_2<-as.data.frame(data_S1L_2)
data_S1L_B<-as.data.frame(data_S1L_B)
names(data_S1L_1)[1]<-"Value"
names(data_S1L_2)[1]<-"Value"
names(data_S1L_B)[1]<-"Value"
data_S1L_1$type<-"MG-132 (PC1)"
data_S1L_2$type<-"Staurosporin (PC2)"
data_S1L_B$type<-"Blank (B)"

new_S1M<-rbind(data_S1L_1, data_S1L_2, data_S1L_B)

#Z score - remember to change dashed line according to threshold
p <- ggplot(new_S1M, aes(x=Value, y=type)) + geom_boxplot() + scale_color_grey() + theme_classic() + coord_flip() + labs(x="Z-factor", y="Variable")  + geom_vline(xintercept=0.3, linetype="dashed", color = "gray", size=2) + xlim(0,0.9)
p


pos<-rbind(data_S1L_1, data_S1L_2)

median(pos$Value)
max(pos$Value) - min(pos$Value)

```


##Figure S1M
```{r}
library(ggplot2)
library(dplyr)
data_s<-read.csv("Input/sandpiper-01_matrix_results.csv")
data_s$Screen<-"Sandpiper1"

data_7<-read.csv("Input/gdsc-007_matrix_results.csv")
data_7$Screen<-"GDSC007"
data_8<-read.csv("Input/gdsc-008_matrix_results.csv")
data_8$Screen<-"GDSC008"
data_9<-read.csv("Input/gdsc-009_matrix_results.csv")
data_9$Screen<-"GDSC009"
data_10<-read.csv("Input/gdsc-010_matrix_results.csv")
data_10$Screen<-"GDSC010b"
data_7plus<-rbind(data_7, data_8, data_9, data_10)

All.data<-rbind(data_s, data_7plus)


#write function to shorten p-values
format_pval <- function(pval){
  pval <- scales::pvalue(pval, accuracy= 0.001, add_p = TRUE)
  gsub(pattern = "(=|<)", replacement = " \\1 ", x = pval)
}


Correlation.BioReps_Average.Responses <- All.data %>%
  filter(CELL_LINE_NAME %in% c("A375", "HT-29", "PC-14", "U-2-OS", "SW620", "C32", "MHH-ES-1")) %>%
  left_join(All.data%>%
              #mutate(Date = as.Date(DATE_CREATED, format = "%d-%b-%y")) %>%
              select(BARCODE,DRUGSET_ID) %>%
              distinct()) %>%
  group_by(DRUGSET_ID,CELL_LINE_NAME, lib1_ID, lib2_ID, Screen) %>%
  summarise(combo_MaxE = mean(combo_MaxE),
            Bliss_matrix = mean(Bliss_matrix),
            Bliss_window = mean(Bliss_window),
            HSA_matrix = mean(HSA_matrix),
            HSA_window = mean(HSA_window)) %>%
  ungroup %>%
  melt(measure.vars = c("combo_MaxE", "Bliss_matrix", "Bliss_window", "HSA_matrix", "HSA_window")) %>%
  full_join(All.data %>%
              filter(CELL_LINE_NAME %in% c("A375", "HT-29", "PC-14", "U-2-OS", "SW620", "C32", "MHH-ES-1")) %>%
              left_join(All.data %>%
                          #mutate(Date = as.Date(DATE_CREATED, format = "%d-%b-%y")) %>%
                          select(BARCODE,DRUGSET_ID)%>%
                          distinct())  %>%
              group_by(DRUGSET_ID,CELL_LINE_NAME, lib1_ID, Screen) %>%
              summarise(lib1_MaxE = mean(lib1_MaxE),
                        lib1_IC50_ln = mean(lib1_IC50_ln)) %>%
              ungroup %>%
              melt(measure.vars = c("lib1_MaxE", "lib1_IC50_ln"))) %>%
  full_join(All.data %>%
              filter(CELL_LINE_NAME %in% c("A375", "HT-29", "PC-14", "U-2-OS", "SW620", "C32", "MHH-ES-1")) %>%
              left_join(All.data%>%
                          #mutate(Date = as.Date(DATE_CREATED, format = "%d-%b-%y")) %>%
                          select(BARCODE,DRUGSET_ID) %>%
                          distinct())  %>%
              group_by(DRUGSET_ID,CELL_LINE_NAME, lib2_ID, Screen) %>%
              summarise(lib2_MaxE = mean(lib2_MaxE),
                        lib2_IC50_ln = mean(lib2_IC50_ln)) %>%
              ungroup %>%
              melt(measure.vars = c("lib2_MaxE", "lib2_IC50_ln"))) %>%
  rename(Metric = variable) %>%
  #arrange(Date) %>%
  group_by(CELL_LINE_NAME, lib1_ID,  lib2_ID, Screen, Metric) %>%
  mutate(Replicate = 1:n()) %>%
  ungroup() %>%
  mutate(Reorder_Metric = case_when(Metric == "lib1_MaxE" ~ 1,
                                    Metric == "lib1_IC50_ln" ~ 2,
                                    Metric == "lib2_MaxE" ~ 3,
                                    Metric == "lib2_IC50_ln" ~ 4,
                                    Metric == "combo_MaxE" ~ 5,
                                    Metric == "Bliss_matrix" ~ 6,
                                    Metric == "Bliss_window" ~ 7,
                                    Metric == "HSA_matrix" ~ 8,
                                    Metric == "HSA_window" ~ 9),
         Metric = case_when(Metric == "lib1_MaxE" ~ "lib1_MaxE",
                            Metric == "lib1_IC50_ln" ~ "lib1_IC50_ln",
                            Metric == "lib2_MaxE" ~ "lib2_MaxE",
                            Metric == "lib2_IC50_ln" ~ "lib2_IC50_ln",
                            Metric == "combo_MaxE" ~ "combo_MaxE",
                            Metric == "Bliss_matrix" ~ "Bliss_matrix",
                            Metric == "Bliss_window" ~ "Bliss_window",
                            Metric == "HSA_matrix" ~ "HSA_matrix",
                            Metric == "HSA_window" ~ "HSA_window")) %>%
  arrange(Reorder_Metric) %>%
  mutate(Metric = factor(Metric, unique(Metric)))


# Calculate Pearson correlation coefficients for each bio rep pair
# There might be easier ways, but...
# full join the above data frame with itself to get all bio rep pairs
# Remove self pairs (e.g. BR1 with BR1)
# Remove duplicated pairs (e.g. BR1/BR2 = BR2/BR1)
# group bio rep pair & metric and calculate pearson correlation coefficient

# DO so by screen: All (join)/Breast/Colon/Pancreas

Correlation.BioReps_Coefficients <- Correlation.BioReps_Average.Responses %>%
  full_join(Correlation.BioReps_Average.Responses %>%
              mutate(Screen = "All")) %>%
  rename(Value_1 = value,
         Replicate_1 = Replicate) %>%
  full_join(Correlation.BioReps_Average.Responses %>%
              full_join(Correlation.BioReps_Average.Responses %>%
                          mutate(Screen = "All"))%>%
              select(-DRUGSET_ID) %>%
              rename(Value_2 = value,
                     Replicate_2 = Replicate)) %>%
  filter(!Replicate_1 == Replicate_2) %>%
  group_by(CELL_LINE_NAME, lib1_ID, lib2_ID, Screen, Metric,
           BioRep_Pair = paste(pmin(Replicate_1, Replicate_2), pmax(Replicate_1, Replicate_2), sep = "_")) %>%
  filter(row_number() == 1) %>%
  ungroup() %>%
  group_by(BioRep_Pair, Metric, Screen) %>%
  mutate(Correlation = cor(Value_1, Value_2, method = "pearson", use = "pairwise.complete.obs"),
         p_value = cor.test(Value_1, Value_2, method = "pearson")$p.value) %>%
  ungroup()



knitr::kable(Correlation.BioReps_Coefficients %>%
               group_by(Screen, Metric) %>%
               summarise(Median_Pearson = round(median(Correlation), digits = 3)) %>%
               ungroup() %>%
               spread(Screen, Median_Pearson), 
             align=c('l', 'l', 'l', 'c', 'c', 'c', 'c', 'c', 'c', 'c'), format="html")%>%
  kable_styling(bootstrap_options=c("stripped","hover","condensed","bordered"), full_width=F, position="left", font_size=12)



# Plot all pearson correlation coefficients by input metric, display as boxplot

P_QC_Correlations.key.metrics <- ggplot(Correlation.BioReps_Coefficients %>%
                                          filter(Screen == "All"),
                                        aes(x = Metric, y = Correlation)) +
  geom_boxplot() +
  theme_bw() +
  ylab("Pearson correlation coefficient") +
  scale_y_continuous(limits = c(0, 1), breaks = seq(0, 1, 0.2)) +
  theme(panel.grid = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 25, hjust = 1),
        axis.title.y = element_text(size = 10)) 

P_QC_Correlations.key.metrics



pdf("S1M_BioReps_Corr of metrics_All data.pdf",
    width = 4, height = 3)
print(P_QC_Correlations.key.metrics)
dev.off()

```

##Figure 2C - N.B. addition annotation added in Adobe Illustrator
```{r}
biomarkers<-read.csv("Input/Overall_ANOVA_results.csv")
#Counting number of rows = number of tests run
nrow(biomarkers)

biomarkers$Analysis<-"Non-significant"

#Subsetting significant biomarkers based on p value, FDR and positive and negative Glass deltas
biomarkers_s<-subset(biomarkers, biomarkers$ANOVA_FEATURE_pval<=0.001)
biomarkers_s<-subset(biomarkers_s, biomarkers_s$ANOVA_FEATURE_FDR<=10)
biomarkers_s<-subset(biomarkers_s, biomarkers_s$FEATURE_neg_Glass_delta>=1)
biomarkers_s<-subset(biomarkers_s, biomarkers_s$FEATURE_pos_Glass_delta>=1)
'%ni%' <- Negate('%in%')

biomarkers_n <-biomarkers[row.names(biomarkers) %ni% row.names(biomarkers_s),]

#Labelling significant biomarkers by analysis type
for (i in 1:nrow(biomarkers_s)){
  if (biomarkers_s$Cancer_type[i]=="All" && biomarkers_s$Genotype[i]=="All"){
    biomarkers_s$Analysis[i] ="Pan-cancer"
  }
  
  if (biomarkers_s$Cancer_type[i]=="All" && biomarkers_s$Genotype[i]!="All"){
    biomarkers_s$Analysis[i] ="Pan-cancer molecular basket"
  }
  
  if (biomarkers_s$Cancer_type[i]!="All"){
    biomarkers_s$Analysis[i] = "By cancer type"
  }
}

biomarkers<-rbind(biomarkers_s, biomarkers_n)
#Sort biomarkers by analysis type
biomarkers<-biomarkers[order(biomarkers$Analysis),]
#volcano plot
p <- ggplot(data =biomarkers, aes(x=FEATURE_delta_MEAN_IC50, y=-log10(ANOVA_FEATURE_pval), col=Analysis)) + geom_point(size=0.5) + theme_light() + xlab("Effect size") + ylab("-log10 ANOVA feature p value") + theme(legend.position="bottom") + (theme(panel.grid = element_blank())) + ylim(0,34) + scale_color_manual(values=c("#df7c73", "#d3d3d3",  "#b49f36", "#5ea74c"))

#pdf(file = "Volcano.pdf")
p
#dev.off()
```

## Figure 2D onwards 
```{r}
full_data<- read.csv("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_biomarkers_with_combo_categories_200622.csv", header = TRUE) 

full_data1<- read.csv("~/OneDrive - AZCollaboration/Projects/Sanger_Data/All_biomarkers_with_combo_categories_200622.csv", header = TRUE) 


#eliminating all combos with no emergent biomarkers

full_data<-full_data[!is.na(full_data$DRUG_ID),]

```

Identify the groups of covariates that are specified in the above general categoies. This is to create groups within categories.

```{r}
ii<-which(full_data$Cancer_type =="All" & full_data$Genotype == "All", arr.ind = TRUE)
full_data$Cancer_type[ii]<-"All-pancancer"
full_data$Genotype[ii]<-"All-pancancer"

iii<-which(full_data$Cancer_type =="All", arr.ind = TRUE)
full_data$Cancer_type[iii]<-"All-molecular baskets"


ftype<-unique(full_data$Features) # biomarker types -- biomarkers captured in FEATURE
ctype<-unique(full_data$Cancer_type) # cancer types and pancancer 
gtype<-unique(full_data$Genotype) # molecular basket and pancancer
catDrug<-unique(full_data$Category_New) # drug category
```

The data contains the following subset of Feature:

```{r}
ftype
```

the following subset of cancer type:

```{r}
ctype
```
Where "All" is pancancer
and the following genotype of cell lines:

```{r}
gtype
catDrug
```

where *All* includes all genotype considered.

Visulisation of the numnber cell lines that have harboured at least one of the features selected, distributed according to these dfferent categories. 

```{r}
library(ggplot2)
library(cowplot)
library(dplyr)

library(ggpubr)
library(colorspace)

library(tidyverse)
library(gprofiler2)
library(tibble)
library(reshape2)

library(pheatmap)

cal_z_score <- function(x){
  (x - mean(x)) / sd(x)
}


```


```{r}
## preparing data structure
plot_ftype<-data.frame(full_data$N_FEATURE_pos, 
                       full_data$FEATURE_IC50_effect_size,
                       full_data$Features, 
                       full_data$Cancer_type,
                       full_data$Genotype,
                       full_data$Category_New)
colnames(plot_ftype)<-c("NumberCL_Present",
                        "IC50_Effect_Size",
                        "Feature_Type", 
                        "Cancer_Type",
                        "Genotype",
                        "Drug_Category")


## Eliminating "All" and "All-pancancer" from tissue types
# since they are not indformtive
temp<-plot_ftype[plot_ftype$Cancer_Type !="All",]
temp<-temp[temp$Cancer_Type !="All-pancancer",]
#temp<-temp[temp$Cancer_Type !="All-molecular baskets",]

plot_ftype<-temp

max_effect<-round(max(full_data$FEATURE_IC50_effect_size))


```

### Comparing biomarkes types over number of cell lines in which they have been detected.

```{r}
# preparing summaries for plotting barchart for Number of cell lines
maxCL<- 
  plot_ftype %>%
  group_by(Feature_Type) %>%
  summarise(grp.mean=mean(NumberCL_Present),
            se=sd(NumberCL_Present)/sqrt(n()))
maxCL$se[is.na(maxCL$se)]<-0

# building the plot
ggplot_ftype<-ggplot(maxCL, aes(x=Feature_Type, y = grp.mean,
                               ymin=grp.mean-se,
                               ymax= grp.mean+se,
                                    fill= Feature_Type))+
  geom_col(width=0.7)+
  geom_errorbar(width=0.25)+
  labs(x= " Biomarkers Type", y = " Average Number of Cell Lines", 
       fill = "Biomarkers Type")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
       legend.position = "right")

ggplot_ftype


maxBN<- 
  plot_ftype %>%
  group_by(Feature_Type) %>%
  summarise(grp.count=length(Feature_Type))
maxBN$grp.count[is.na(maxBN$grp.count)]<-0


# building the plot for counting the biomarkers
ggplot_btype<-ggplot(maxBN, aes(x=Feature_Type, y = grp.count, fill= Feature_Type))+
  geom_col(width=0.7)+
  labs(x= " Biomarkers Type", y = " Number of Biomarkers", 
       fill = "Biomarkers Type")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
       legend.position = "right")

ggplot_btype
maxBN
```

```{r}
# Preparing summaries to plot IC50 by groups

# plotting IC50 by grouos
IC50es<- 
  plot_ftype %>%
  group_by(Feature_Type) %>%
  summarise(grp.mean=mean(IC50_Effect_Size),
            se=sd(IC50_Effect_Size)/sqrt(n()))

ggplot_ftypeIC50<-ggplot(IC50es, aes(x=Feature_Type, y = grp.mean,
                               ymin=grp.mean-se,
                               ymax= grp.mean+se,
                               fill= Feature_Type))+
  ylim(0,max_effect)+
  geom_col(width=0.7)+
  geom_errorbar(width=0.25)+
  labs(x= " Biomarkers Type", y = " Average Y score", 
       fill = "Biomarkers Type")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
       legend.position = "right")


ggplot_ftypeIC50


##using box plots
plot_ftype %>% tidyr::gather("id", "value",2) %>% 
ggplot(.,aes(x=Feature_Type, y=value, color= Feature_Type, fill=after_scale(desaturate(lighten(color, .6), .6)), shape=Feature_Type))+
  geom_boxplot(alpha = 0.80, outlier.shape=NA) + 
  ylim(0,5)+
  geom_jitter(width = .3, aes(fill=Feature_Type), shape = 21, color = "black")+ 
  labs(x=" Biomarkers Type", y= 'Input - Effect Size ') +
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
       legend.position = "right") +
  facet_wrap(~id)

```


From this first visualisation it is possible to see that although the *mutation -AZ* biomarker caries the largest effect size, it also has the largest standard error, which is an indicatore of low precision. The *mutation - Sanger* is better in precision but with less potency that *CNA- AZ* which has the over better IC50 effect size and good precision.  

### Comparing biomarkes types over cancer type using Y effect size


```{r}
# preparing summaries for plotting barchart for Number of cell lines
maxCT<- 
  plot_ftype %>%
  group_by(Cancer_Type) %>%
  summarise(grp.mean=mean(IC50_Effect_Size),
            se=sd(IC50_Effect_Size)/sqrt(n()))
maxCT$se[is.na(maxCT$se)]<-0



# building the plot
ggplot_ctype<-ggplot(maxCT, aes(x=Cancer_Type, y = grp.mean,
                                ymin=grp.mean-se,
                                ymax= grp.mean+se,
                                fill= Cancer_Type))+
  geom_col(width=0.7)+
  ylim(0,max_effect)+
  geom_errorbar(width=0.25)+
  labs(x= " Cancer Type", y = " Average Y score", 
       fill = "Cancer Type")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
        legend.position = "right")

ggplot_ctype

png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/IC50_effect_Tisue.png", res = 150, width =840,height = 840)
##using box plots

## eliminaty All-molecular Baskets to make plot clear
plot_ftype1<- plot_ftype[plot_ftype$Cancer_Type!="All-molecular baskets",]

plot_ftype1 %>% tidyr::gather("id", "value",2) %>% 
ggplot(.,aes(x=Cancer_Type, y=value, color= Cancer_Type, fill=after_scale(desaturate(lighten(color, .6), .6)), shape=Cancer_Type))+
  geom_boxplot(alpha = 0.80, outlier.shape=NA) + 
  ylim(0,5)+
  geom_jitter(width = .3, aes(fill=Cancer_Type), shape = 21, color = "black")+ 
  labs(x=" Cancer Type", y= 'Input - Effect Size ') +
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
       legend.position = "right") +
  facet_wrap(~id)
dev.off()

```

From this plot it is possible to visualise that there will be a small set of feature that have a *large* Y score on all type of cancer. However, on different types of cancer the compounds used in the combos have different effect. The largest effect are possibly on disease segment that we are more specialised in. Interesting is to see the pancreatic cancer and the Ewing's sarcoma appearing almosts the top hits.  

### Comparing Genotype with  Y effect size

## Genotype comparison.

```{r}

maxGT<- 
  plot_ftype %>%
  group_by(Genotype) %>%
  summarise(grp.mean=mean(IC50_Effect_Size),
            se=sd(IC50_Effect_Size)/sqrt(n()))
maxCT$se[is.na(maxCT$se)]<-0

# building the plot
ggplot_gtype<-ggplot(maxGT, aes(x=Genotype, y = grp.mean,
                                ymin=grp.mean-se,
                                ymax= grp.mean+se,
                                fill= Genotype))+
  geom_col(width=0.7)+
  ylim(0,max_effect)+
  geom_errorbar(width=0.25)+
  labs(x= " Genotype", y = "Average  Y score", 
       fill = "Genotype")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
        legend.position = "right")

ggplot_gtype

##using box plots

png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/IC50_effect_Basket.png", res = 150,width =840,height = 840)

## renaming All in Genotype as All-Genotypes
plot_ftype1<-plot_ftype
plot_ftype1$Genotype[plot_ftype1$Genotype=="All"]<-"All-Genotypes"

plot_ftype1 %>% tidyr::gather("id", "value",2) %>% 
ggplot(.,aes(x=Genotype, y=value, color= Genotype, fill=after_scale(desaturate(lighten(color, .6), .6)), shape=Genotype))+
  geom_boxplot(alpha = 0.80, outlier.shape=NA) + 
  ylim(0,5)+
  geom_jitter(width = .3, aes(fill=Genotype), shape = 21, color = "black")+ 
  labs(x=" Molecular Baskets", y= 'Input - Effect Size ') +
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
       legend.position = "right") +
  facet_wrap(~id)
dev.off()


maxDC<- 
  plot_ftype %>%
  group_by(Drug_Category) %>%
  summarise(grp.mean=mean(IC50_Effect_Size),
            se=sd(IC50_Effect_Size)/sqrt(n()))
maxDC$se[is.na(maxDC$se)]<-0

# building the plot
ggplot_gtype<-ggplot(maxDC, aes(x=Drug_Category, y = grp.mean,
                                ymin=grp.mean-se,
                                ymax= grp.mean+se,
                                fill= Drug_Category))+
  geom_col(width=0.7)+
  ylim(0,max_effect)+
  geom_errorbar(width=0.25)+
  labs(x= " Drug Category", y = "Average  Y score", 
       fill = "Drug_Category")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
        legend.position = "right")

ggplot_gtype

##using box plots

png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/IC50_effect_Drug_Category.png", res = 150,width =1140,height = 840)
plot_ftype %>% tidyr::gather("id", "value",2) %>% 
ggplot(.,aes(x=Drug_Category, y=value, color= Drug_Category, fill=after_scale(desaturate(lighten(color, .6), .6)), shape=Drug_Category))+
  geom_boxplot(alpha = 0.80, outlier.shape=NA) + 
  ylim(0,5)+
  geom_jitter(width = .3, aes(fill=Drug_Category), shape = 21, color = "black")+ 
  labs(x=" Drug Category", y= 'Input - Effect Size ') +
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
       legend.position = "right") +
  facet_wrap(~id)
dev.off()








```

This barchart gives you a complete overview of the final biomarkers selected when grouped by genotype. It is interesting to note that the standard error is rather small and comparable acroos the data sets. 

## Evaluating the impact of the covariates in selection of biomarkers 

In order to see how the list of selected biomarkes was influenced by the covariates, we will plot a barchart of $-log_{10}(pval)$ of the Y effect size derived by the ANOVA model grouped by the different biomarkers type 

```{r}
# data preparation
pvalBT<- 
  full_data %>%
  group_by(Features) %>%
  summarise(grp.mean=mean(-log10(ANOVA_FEATURE_pval)),
            grp.max=max(-log10(ANOVA_FEATURE_pval)),
            grp.npval=length(ANOVA_FEATURE_pval),
            se=sd(-log10(ANOVA_FEATURE_pval))/sqrt(n()))
pvalBT$se[is.na(pvalBT$se)]<-0
```

```{r}
# plotting the mean log pvalues

ggplot_pvalbt<-ggplot(pvalBT, aes(x=Features, y = grp.mean,
                                ymin=grp.mean-se,
                                ymax= grp.mean+se,
                                fill= Features))+
  geom_col(width=0.7)+
  geom_errorbar(width=0.25)+
  labs(x= " Biomarkers Type", y = " Average -log10(pval Y score)", 
       fill = "Genotype")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
        legend.position = "right")

ggplot_pvalbt
```

```{r}
## looking at each distribution 

# data preparation
pvalBTDis<- 
  full_data %>%
  group_by(Features)%>%
  summarise(grp.data=-log10(ANOVA_FEATURE_pval))

#plotting data
ggplot_pvalbtDIS<-ggplot(pvalBTDis, aes(x=Features, y = grp.data,
                                fill= Features))+
  geom_boxplot()+
  labs(x= " Biomarkers Type", y = " Average -log10(pval Y score)", 
       fill = "Genotype")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
        legend.position = "right")

ggplot_pvalbtDIS
```

The distrubutions of the mean valuse is fairy uniform and the error bars are quite small a part from the *mutation AZ* and *Mutation - Sanger* which we have already detected are rather noisy. What is interesting is that if we look at the number of the pvalues that have been selected for each fature, we have the following scenario:

```{r}
ggplot_pvalbt1<-ggplot(pvalBT, aes(x=Features, y = grp.max,
                                  fill= Features))+
  geom_col(width=0.7)+
  labs(x= " Biomarkers Type", y = " Max -log10(pval Y Score)", 
       fill = "Biomarker Type")+
  theme(text = element_text(size=12),
        axis.text.x = element_text(angle = 70, vjust = 1, hjust=1),
        legend.position = "right")

ggplot_pvalbt1
```

where the gene expression have a huge bias in the model in terms of selection for final hits.It might also be worth noting that over 1200 of hits are gene of the expression type.
```{r}
pvalBT
```


### Evaluating impact of Drugs combinations on Biomarkes types and Cancer types

We are using the heat maps and the hierarchical clustering with Euclidean distance as a similarity function to visualise the Y combinations score acreoss the drug combinations selected. 


We will look at the Y scores for cell lines habouring the features and the delta mean Y score to highlight sentitivity and Resistance. 

```{r}
#full_data$Cancer_type

# building the data structure
library(reshape2)

cBT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Features) %>%
  summarise(grp.mean=mean(FEATURE_IC50_effect_size))

temp<-dcast(cBT, DRUG_NAME.x ~ Features)
comboBT<-temp
```


```{r}
# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboBT1<-comboBT
comboBT1[is.na(comboBT1)]<-0
row.names(comboBT1)<- comboBT1[,1]

comboBT1 <- comboBT1[,2:5]

```

```{r}

pheatmap(as.matrix(comboBT1), cluster_cols = TRUE, 
         fontsize_row = 4, fontsize_col = 7)



```


```{r}
#### ALternative way to draw heat,map with correlation as similarity

# Pairwise correlation between samples (columns)
cols.cor <- cor(comboBT1, use = "pairwise.complete.obs", method = "pearson")
# Pairwise correlation between rows (genes)
rows.cor <- cor(t(comboBT1), use = "pairwise.complete.obs", method = "pearson")

## Row- and column-wise clustering using correlation 
hclust.col <- hclust(as.dist(1-cols.cor)) 
hclust.row <- hclust(as.dist(1-rows.cor))


# Plot the heatmap

pheatmap(as.matrix(comboBT1), cluster_cols = TRUE, 
         fontsize_row = 4, fontsize_col = 7)


```

The heatmap is built using 1 minus pearson correlation as similarity measure. From this Heatmap it is evident how the effect of the Y score can be associated to one biomaker type or more for each combination evaluated. In the heatmap we have four clusters of combinations that are formed based on their Y score evaluated with the ANOVA model.

Looking at correlation matrices for the interactions across feature types.

```{r}
library(gplots)
# heatmap with correlation
symnum( cU <- cor(comboBT1) )

 ## The Correlation matrix with same reordering:
 hM <- format(round(cU, 2))


 # now with the correlation matrix on the plot itself

 heatmap.2(cU, Rowv=FALSE, symm=TRUE, col=rev(heat.colors(16)),
             distfun=function(c) as.dist(1 - c), trace="none",
             cellnote=hM, margins=c(12,12))
```

Evaluating the correlation of teh cancer types with the Y score  

```{r}
# building the data structure for cancer type
library(reshape2)

cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Cancer_type) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(cCT, DRUG_NAME.x ~ Cancer_type)
comboCT<-temp
```

```{r}
# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:14]

# Eliminating rows with all zeros. 
#id<-which(rowSums(comboCT1)==0,arr.ind=TRUE)
#comboCT1<-comboCT1[-id,]



```

Drawing heatmap using Euclidean distance  correlation as similarity

```{r}

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, 
         fontsize_row = 4, fontsize_col = 7)


```
Using molecular baskets

```{r}
 cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Genotype) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(cCT, DRUG_NAME.x ~ Genotype)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:8]


# Drawing heatmap using Euclidean distance  correlation as similarity

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, 
         fontsize_row = 4, fontsize_col = 7)
```
Using drug categories
```{r}
 cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Category_New) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(cCT, DRUG_NAME.x ~ Category_New)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:7]


# Drawing heatmap using Euclidean distance  correlation as similarity

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, 
         fontsize_row = 4, fontsize_col = 7)
```


## Grouping by feature type

```{r}
cCT<- 
  full_data %>%
  group_by(Features,Genotype) %>%
  summarise(grp.count=length(Features))

temp<-dcast(cCT, Genotype ~ Features)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:5]


# Drawing heatmap using Euclidean distance  correlation as similarity
png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Biomarkers_summary_Baskets.png", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE, 
         fontsize_row = 7, fontsize_col = 7, display_numbers = as.matrix(comboCT1))
dev.off()


png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Biomarkers_summary_Baskets_notAll.png", res = 150)

## removing All and All-pancancer for clarity in figures
comboCT2<-comboCT1[!rownames(comboCT1)=="All",]
comboCT2<-comboCT2[!rownames(comboCT2)=="All-pancancer",]

pheatmap(as.matrix(comboCT2), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, display_numbers = as.matrix(comboCT2))
dev.off()
```
### Grouping by Cancer Type

```{r}
cCT<- 
  full_data %>%
  group_by(Features,Cancer_type) %>%
  summarise(grp.count=length(Features))

temp<-dcast(cCT, Cancer_type ~ Features)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:5]


# Drawing heatmap using Euclidean distance  correlation as similarity
png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Biomarkers_summary_Tisue.png", res = 150,width= 680)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, cellwidth = 20, display_numbers = as.matrix(comboCT1))

dev.off()

## eliminating pancancer from plot to enhance clarity

png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Biomarkers_summary_Tisue_notAll.png", res = 150,width= 680)
## removing All and All-pancancer for clarity in figures 
comboCT2<-comboCT1[!rownames(comboCT1)=="All",]
comboCT2<-comboCT2[!rownames(comboCT2)=="All-pancancer",]

pheatmap(as.matrix(comboCT2), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, cellwidth = 20, display_numbers = as.matrix(comboCT2))
dev.off()

png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Biomarkers_summary_Tisue_notAll_notMolBaskets.png", res = 150,width= 680)
## removing All and All-pancancer for clarity in figures 
comboCT2<-comboCT1[!rownames(comboCT1)=="All",]
comboCT2<-comboCT2[!rownames(comboCT2)=="All-pancancer",]
comboCT2<-comboCT2[!rownames(comboCT2)=="All-molecular baskets",]

pheatmap(as.matrix(comboCT2), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, cellwidth = 20, display_numbers = as.matrix(comboCT2))
dev.off()
```
## Grouping by categories
```{r}
cCT<- 
  full_data %>%
  group_by(Features,Category_New) %>%
  summarise(grp.count=length(Features))

temp<-dcast(cCT, Category_New ~ Features)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:5]


# Drawing heatmap using Euclidean distance  correlation as similarity
png(filename = "~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Biomarkers_summary_Categories.png", res = 150, width = 880, height = 680)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, cellwidth = 20, cellheight = 15,
         display_numbers = as.matrix(comboCT1))
dev.off()
```


#### Grouping by Drug names
```{r}

# select the final number of combination before doing this.
# Loading the pathways and defining the final 113 combos for plotting

final_combos_path<- read.csv("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Combo pathways.csv", header = TRUE)

idx<- match(final_combos_path$COMBO_ID,full_data$DRUG_ID, nomatch = 0) #final 113 combos

#final set of combination 113 in total
final_data<-full_data[idx,]



cCT<- 
  final_data %>%
  group_by(Features,DRUG_NAME.x) %>%
  summarise(grp.count=length(Features))

temp<-dcast(cCT, DRUG_NAME.x ~ Features)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:4]
comboCT1_counts<-comboCT1

# Drawing heatmap using Euclidean distance  correlation as similarity

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/DRUG_Name_Patways_Biomarkers.png",width= 2680, height = 2680, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, display_numbers = as.matrix(comboCT1),cellwidth = 35, cellheight = 10)
dev.off()



### repeating plots for effect size

cCT<- 
  final_data %>%
  group_by(Features,DRUG_NAME.x) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(cCT, DRUG_NAME.x ~ Features)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:4]


# Drawing heatmap using Euclidean distance  correlation as similarity
png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Drug_Name_Patways_EffectSize.png",width= 2680, height = 2860, units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, cellwidth = 35, cellheight = 10, 
         display_numbers = as.matrix(comboCT1_counts))
dev.off()
```
### Plotting pathways vs number of Biomarkers and Effect size.
```{r}

## finding the pathways to add to the data table
idx1<- match(final_data$DRUG_ID,final_combos_path$COMBO_ID, nomatch = 0)

final_path<-final_combos_path$Combo_pathway[idx1]

final_data_path<-cbind(final_data,final_path)

cCT<- 
  final_data_path %>%
  group_by(Features,final_path) %>%
  summarise(grp.count=length(Features))

temp<-dcast(cCT, final_path ~ Features)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:4]
comboCT1_counts<-comboCT1

# Drawing heatmap using Euclidean distance  correlation as similarity
png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Pathways_Biomarkers.png",width= 880, height = 880, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, display_numbers = as.matrix(comboCT1), cellwidth = 30, cellheight = 8 )
dev.off()


#pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
#         fontsize_row = 7, fontsize_col = 7, display_numbers = as.matrix(comboCT1),cellwidth = 35, cellheight = 10 )



### plotting pathway and effect size
cCT<- 
  final_data_path %>%
  group_by(Features,final_path) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(cCT, final_path ~ Features)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:4]


# Drawing heatmap using Euclidean distance  correlation as similarity
png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Pathways_Effectsize.png",
    width= 880, height = 880,units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, cellwidth = 30, cellheight = 8, 
         display_numbers = as.matrix(comboCT1_counts) )
dev.off()


```
### Pathways for categories
```{r}

## finding the pathways to add to the data table
idx1<- match(final_data$DRUG_ID,final_combos_path$COMBO_ID, nomatch = 0)

final_path<-final_combos_path$Combo_pathway[idx1]

final_data_path<-cbind(final_data,final_path)

cCT<- 
  final_data_path %>%
  group_by(Category_New,final_path) %>%
  summarise(grp.count=length(Category_New))

temp<-dcast(cCT, final_path ~ Category_New)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:10]
comboCT1_counts<-comboCT1

# Drawing heatmap using Euclidean distance  correlation as similarity
png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Pathways_Categories.png",
    width= 1280, height = 1280,units = "px", res = 150)

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, display_numbers = as.matrix(comboCT1), cellwidth = 30, cellheight = 8 )
dev.off()


#pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
#         fontsize_row = 7, fontsize_col = 7, display_numbers = as.matrix(comboCT1),cellwidth = 35, cellheight = 10 )



### plotting pathway and effect size
cCT<- 
  final_data_path %>%
  group_by(Category_New,final_path) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(cCT, final_path ~ Category_New)
comboCT<-temp


# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:10]


# Drawing heatmap using Euclidean distance  correlation as similarity
png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Pathways_Category_Effect_size.png",width= 1280, height = 1280,units = "px", res = 150)

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 7, cellwidth = 30, cellheight = 8, 
         display_numbers = as.matrix(comboCT1_counts) )
dev.off()


```







```{r}
# heatmap with correlation across cancer type
symnum( cU <- cor(comboCT1) )

 ## The Correlation matrix with same reordering:
 hM <- format(round(cU, 2))


 # now with the correlation matrix on the plot itself

 heatmap.2(cU, Rowv=FALSE, symm=TRUE, col=rev(heat.colors(16)),
             distfun=function(c) as.dist(1 - c), trace="none",
             cellnote=hM, cexRow= 0.5, cexCol = 0.5, margins=c(6,6))
```


```{r}
# Looking at sensitivity and Resistance using the delta mean

cBT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Features) %>%
  summarise(grp.mean=mean(FEATURE_delta_MEAN_IC50))

temp<-dcast(cBT, DRUG_NAME.x ~ Features)
comboBT<-temp

```

```{r}
# setting NA as zero to plot a heatmap and HC of the data
# setting NA as zero for hc and heatmap
comboBT1<-comboBT
comboBT1[is.na(comboBT1)]<-0
row.names(comboBT1)<- comboBT1[,1]

comboBT1 <- comboBT1[,2:5]
```


```{r}

pheatmap(as.matrix(comboBT1), cluster_cols = FALSE, 
         fontsize_row = 4, fontsize_col = 7)


```

From the above heatmap it is clear where the strongest resistance interactions (green) and the sensitivity interactions are distributed for features. It would be also interesting to explore this aspect by grouping for cancer types. 

```{r}
### looking at Sensitivity and Resistance cancer type
cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Cancer_type) %>%
  summarise(grp.median=median(FEATURE_delta_MEAN_IC50))

temp<-dcast(cCT, DRUG_NAME.x ~ Cancer_type)
comboCT<-temp


# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]



comboCT1 <- comboCT1[,2:14]
## removing All and All-pancancer for clarity in figures 
comboCT1<-comboCT1[!rownames(comboCT1)=="All",]
comboCT1<-comboCT1[!rownames(comboCT1)=="All-pancancer",]


### Counting the contribution to the median
cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Cancer_type) %>%
  summarise(grp.counts=length(FEATURE_delta_MEAN_IC50))

temp<-dcast(cCT, DRUG_NAME.x ~ Cancer_type)
comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT2<-comboCT
comboCT2[is.na(comboCT2)]<-0
row.names(comboCT2)<- comboCT2[,1]

comboCT1_counts <- comboCT2[,2:14]




png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Drug_Sensitivity.png",width= 2680, height = 2680, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 8, cellwidth = 30, cellheight = 8, 
         display_numbers = as.matrix(comboCT1_counts) )
dev.off()


## molecular baskets
cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Genotype) %>%
  summarise(grp.median=median(FEATURE_delta_MEAN_IC50))

temp<-dcast(cCT, DRUG_NAME.x ~ Genotype)
comboCT<-temp

## removing All and All-pancancer for clarity in figures 
comboCT<-comboCT[,!colnames(comboCT)=="All"]
comboCT<-comboCT[,!colnames(comboCT)=="All-pancancer"]

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:7]


### Counting the contribution to the median
cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Genotype) %>%
  summarise(grp.counts=length(FEATURE_delta_MEAN_IC50))

temp<-dcast(cCT, DRUG_NAME.x ~ Genotype)
comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT2<-comboCT
comboCT2[is.na(comboCT2)]<-0
row.names(comboCT2)<- comboCT2[,1]

comboCT1_counts <- comboCT2[,2:7]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Genotype_Sensitivity.png",
    width= 1880, height = 1880, units = "px", res = 150)

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 8, cellwidth = 30, cellheight = 8, 
         display_numbers = as.matrix(comboCT1_counts), angle_col = 45 )
dev.off()


## Drug Categories
cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Category_New) %>%
  summarise(grp.median=median(FEATURE_delta_MEAN_IC50))

temp<-dcast(cCT, DRUG_NAME.x ~ Category_New)
comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:10]
comboCT1_counts<-comboCT1


#### counting contribution to the median
cCT<- 
  full_data %>%
  group_by(DRUG_NAME.x,Category_New) %>%
  summarise(grp.counts=length(FEATURE_delta_MEAN_IC50))

temp<-dcast(cCT, DRUG_NAME.x ~ Category_New)
comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT2<-comboCT
comboCT2[is.na(comboCT2)]<-0
row.names(comboCT2)<- comboCT2[,1]

comboCT2 <- comboCT2[,2:10]
comboCT1_counts<-comboCT2

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Category_Sensitivity.png",width= 1880, 
    height = 1880, units = "px", res = 150)

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 7, fontsize_col = 8, cellwidth = 30, cellheight = 8, 
         display_numbers = as.matrix(comboCT1_counts),angle_col = 45)
dev.off()


```

In the heatmap you can note that the majority of the gene expression markers are generally associate to all type of cancers, few are assocated to Lung cancer and others are quite specific. Shared biomarkers might be an interesting follow up. 

The correlation analysis does not show a strong correlation between cancer types based on the effect sizes of the drug combinations. There is a slight positive correlation for the Methylation features and the genomic mutation, which might be expected. The copy Number Alteration from AZ and Sanger databases are also correlated as expected.  

The associations of the biomarkers with the drug combination is not clear if can be detected using the linear model described above. 


### Split the inibition effect ( comboEmax) input from Synergistic (Bliss/HSA)


```{r}
comboE<-full_data[which(full_data$Input=="Combo Emax", arr.ind=TRUE),]
bliss<-full_data[which(full_data$Input=="Bliss matrix", arr.ind=TRUE),]

test<-merge(comboE,bliss,by.x="DRUG_NAME.x", by.y="DRUG_NAME.x", all=TRUE)
#write.csv(test,file="merge_BLiss_ComboE.csv")

```

```{r}
## heatmap for ComboEmax INPUT ( INIBITION effect)
# Y score

ccCT<- 
  comboE %>%
  group_by(DRUG_NAME.x,Features) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(ccCT, DRUG_NAME.x ~ Features)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:5]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Emax_Effect size.png",width= 2680, height = 2680, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, cellwidth = 35,cellheight = 10)
dev.off()




#### counting biomarkers

ccCT<- 
  comboE %>%
  group_by(DRUG_NAME.x,Features) %>%
  summarise(grp.count=length(Features))

temp<-dcast(ccCT, DRUG_NAME.x ~ Features)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:5]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Emax_Biomarkers.png",width= 2480, height = 2480, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, display_numbers =as.matrix(comboCT1), cellwidth = 35,cellheight = 10)
dev.off()

### category describition
ccCT<- 
  comboE %>%
  group_by(DRUG_NAME.x,Category_New) %>%
  summarise(grp.median=median(FEATURE_IC50_effect_size))

temp<-dcast(ccCT, DRUG_NAME.x ~ Category_New)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:10]
comboCT1_counts<-comboCT1

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/category_Emax_Effect size.png",
    width= 2680, height = 2680, units = "px", res = 150)

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, cellwidth = 35,cellheight = 10, 
         display_numbers = as.matrix(comboCT1_counts))
dev.off()




#### counting biomarkers

ccCT<- 
  comboE %>%
  group_by(DRUG_NAME.x,Category_New) %>%
  summarise(grp.count=length(Category_New))

temp<-dcast(ccCT, DRUG_NAME.x ~ Category_New)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:10]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Emergent_images/Emax_Category.png",width= 2480, height = 2480, 
    units = "px", res = 150)

pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, display_numbers =as.matrix(comboCT1), cellwidth = 35,cellheight = 10)

dev.off()




```

# BLISS data

```{r}
ccCT<- 
  bliss %>%
  group_by(DRUG_NAME.x,Features) %>%
  summarise(grp.mean=mean(FEATURE_IC50_effect_size))

temp<-dcast(ccCT, DRUG_NAME.x ~ Features)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:4]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Bliss_Effect size.png",width= 2680, height = 2680, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, cellwidth = 35,cellheight = 10)
dev.off()




#### counting biomarkers

ccCT<- 
  bliss %>%
  group_by(DRUG_NAME.x,Features) %>%
  summarise(grp.count=length(Features))

temp<-dcast(ccCT, DRUG_NAME.x ~ Features)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:4]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Bliss_Biomarkers.png",width= 2480, height = 2480, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, display_numbers =as.matrix(comboCT1), cellwidth = 35,cellheight = 10)
dev.off()

#### cdrug category 


ccCT<- 
  bliss %>%
  group_by(DRUG_NAME.x,Category_New) %>%
  summarise(grp.mean=mean(FEATURE_IC50_effect_size))

temp<-dcast(ccCT, DRUG_NAME.x ~ Category_New)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:7]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/category_Bliss_Effect size.png",width= 2680, height = 2680, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, cellwidth = 35,cellheight = 10)
dev.off()

### counting categories

ccCT<- 
  bliss %>%
  group_by(DRUG_NAME.x,Category_New) %>%
  summarise(grp.count=length(Category_New))

temp<-dcast(ccCT, DRUG_NAME.x ~ Category_New)


comboCT<-temp

# setting NA as zero for hc and heatmap
comboCT1<-comboCT
comboCT1[is.na(comboCT1)]<-0
row.names(comboCT1)<- comboCT1[,1]

comboCT1 <- comboCT1[,2:7]

png("~/OneDrive - AZCollaboration/Projects/Sanger_Data/Bliss_Category.png",width= 2480, height = 2480, 
    units = "px", res = 150)
pheatmap(as.matrix(comboCT1), cluster_cols = FALSE, cluster_rows = FALSE,
         fontsize_row = 6, fontsize_col = 7, display_numbers =as.matrix(comboCT1), cellwidth = 35,cellheight = 10)
dev.off()



```



## Evaluation of how well the features selected can separate the Cancer types and genotypes.


```{r}
library(Rtsne)

# preparing teh data
vis_mat<- 
  full_data %>%
  group_by(FEATURE,DRUG_NAME.x,Cancer_type,Genotype)%>%
  summarise(grp.mean=mean(FEATURE_IC50_effect_size))


temp<-dcast(vis_mat,DRUG_NAME.x~ Genotype,
            fun.aggregate = mean, value.var = "grp.mean")


mat<-t(temp[,2:8])
colnames(mat)<-temp[,1]

mat[is.nan(mat)]<-0

# running tSNE
tsne_mat<-Rtsne(mat,labels=as.factor(row.names(mat)), perplexity=2)
plot_mat<- data.frame(tsne_x=tsne_mat$Y[,1],tsne_y=tsne_mat$Y[,2], 
                      rownames(mat))

colnames(plot_mat)[3]<-"Genotype"

ggp1<-ggplot(plot_mat,aes(x=tsne_x,y=tsne_y,color=Genotype))+
  geom_point()

ggp1

```

The data is clearly separated by genotype using the feature extracted and gruped by drugs compounds. It is uqually applicable for cancer types, as shawn in the plot below.

```{r}
temp<-dcast(vis_mat,DRUG_NAME.x~ Cancer_type,
            fun.aggregate = mean, value.var = "grp.mean")


mat<-t(temp[,2:9])
colnames(mat)<-temp[,1]

mat[is.nan(mat)]<-0

# running tSNE
tsne_mat<-Rtsne(mat,labels=as.factor(row.names(mat)), perplexity=2)
plot_mat<- data.frame(tsne_x=tsne_mat$Y[,1],tsne_y=tsne_mat$Y[,2], 
                      rownames(mat))

colnames(plot_mat)[3]<-"Cancer_Type"

ggp2<-ggplot(plot_mat,aes(x=tsne_x,y=tsne_y,color=Cancer_Type))+
  geom_point()

ggp2
```

The tSNE plots clearly show that the cancer types can be separated based on the interaction effect of the drugs combinations. There is no distinct separation of the genotype which can make laed to the hypothesis that the mayor impact in this selection is palyed by the interactions of the combinations the drugs within the cancer types. 



##Figure S2A
```{r}
biomarkers_sig<-read.csv("Input/Overall_ANOVA_results_significant_large_effect_10_FDR_cutoff.csv")
biomarkers_sig$list<-paste(biomarkers_sig$DRUG_ID, biomarkers_sig$FEATURE, biomarkers_sig$Features, biomarkers_sig$Input, sep="_")

biomarkers_sig_pancan<-subset(biomarkers_sig, biomarkers_sig$Cancer_type=="All"&biomarkers_sig$Genotype=="All")
biomarkers_sig_pancan<-biomarkers_sig_pancan[,-1:-27]

biomarkers_sig_percan<-subset(biomarkers_sig,biomarkers_sig$Genotype=="All"&&biomarkers_sig$Cancer_type!="All")
biomarkers_sig_percan<-biomarkers_sig_percan[,-1:-27]

length(intersect(biomarkers_sig_pancan, biomarkers_sig_percan))
length(setdiff(biomarkers_sig_pancan, biomarkers_sig_percan))
length(setdiff(biomarkers_sig_percan, biomarkers_sig_pancan))
```

##Figure 2B
```{r}
biomarkers_sig_basket<-subset(biomarkers_sig, biomarkers_sig$Genotype!="All")
biomarkers_sig_basket<-biomarkers_sig_basket[,-1:-27]

length(intersect(biomarkers_sig_pancan, biomarkers_sig_basket))
length(setdiff(biomarkers_sig_pancan, biomarkers_sig_basket))
length(setdiff(biomarkers_sig_basket, biomarkers_sig_pancan))
```

##Figure S2C
```{r}
colobreast<-read.csv("Biomarker_inputs/GDSC010_Bliss_window_matrix.csv")
mobem<-read.csv("Biomarker_inputs/COSMIC_KRAS_binarised_gene_expression_subset.csv")
data1<-merge(colobreast, mobem, by="COSMIC_ID")
library(ggplot2)
data1$PIK3CG_up<-gsub("1", "Elevated PIK3CG, n=3", data1$PIK3CG_up)
data1$PIK3CG_up<-gsub("0", "Non-elevated PIK3CG, n=53", data1$PIK3CG_up)

data1$PIK3CG_up<-as.factor(data1$PIK3CG_up)
data2<-data1
p <- ggplot(data2, aes(x=data2$PIK3CG_up, y=data2$X24932545, colour= data2$PIK3CG_up, fill=data2$PIK3CG_up)) + theme_bw() +
  geom_boxplot(alpha = 0.2) +
  geom_jitter(alpha = 0.75, width = 0.2) +
  guides(colour = F, fill = F) +
  theme(panel.grid = element_blank()) + labs(title="",x="", y = "AZD8186 + palbociclib Bliss window") + theme(text = element_text(size = 18))

p

```

##Figure S2D
```{r}
colobreast<-read.csv("Biomarker_inputs/GDSC010_lib2_MaxE_matrix.csv")
mobem<-read.csv("Biomarker_inputs/COSMIC_KRAS_binarised_gene_expression_subset.csv")
data1<-merge(colobreast, mobem, by="COSMIC_ID")
library(ggplot2)
data1$BCL2L11_up<-gsub("1", "Elevated BCL2L11, n=3", data1$BCL2L11_up)
data1$BCL2L11_up<-gsub("0", "Non-elevated BCL2L11, n=53", data1$BCL2L11_up)

data1$BCL2L11_up<-as.factor(data1$BCL2L11_up)
data2<-data1
p <- ggplot(data2, aes(x=data2$BCL2L11_up, y=data2$X24342495, colour= data2$BCL2L11_up, fill=data2$BCL2L11_up)) + theme_bw() +
  geom_boxplot(alpha = 0.2) +
  geom_jitter(alpha = 0.75, width = 0.2) +
  guides(colour = F, fill = F) +
  theme(panel.grid = element_blank()) + labs(title="",x="", y = "AZD4320 Emax") + theme(text = element_text(size = 18))

p
```

##Figure S2E
```{r}
colobreast<-read.csv("Biomarker_inputs/GDSC010_lib2_MaxE_matrix.csv")
mobem<-read.csv("Biomarker_inputs/COSMIC_KRAS_binarised_gene_expression_subset.csv")
data1<-merge(colobreast, mobem, by="COSMIC_ID")
library(ggplot2)
data1$BCL2L11_up<-gsub("1", "Elevated BCL2L11, n=3", data1$BCL2L11_up)
data1$BCL2L11_up<-gsub("0", "Non-elevated BCL2L11, n=53", data1$BCL2L11_up)

data1$BCL2L11_up<-as.factor(data1$BCL2L11_up)
data2<-data1
p <- ggplot(data2, aes(x=data2$BCL2L11_up, y=data2$X24342497, colour= data2$BCL2L11_up, fill=data2$BCL2L11_up)) + theme_bw() +
  geom_boxplot(alpha = 0.2) +
  geom_jitter(alpha = 0.75, width = 0.2) +
  guides(colour = F, fill = F) +
  theme(panel.grid = element_blank()) + labs(title="",x="", y = "Venetoclax Emax") + theme(text = element_text(size = 18))

p

```

##Figure S2F
```{r}
colobreast<-read.csv("Biomarker_inputs/GDSC010_lib2_MaxE_matrix.csv")
mobem<-read.csv("Biomarker_inputs/COSMIC_TP53_binarised_gene_expression_subset.csv")
data1<-merge(colobreast, mobem, by="COSMIC_ID")
library(ggplot2)
data1$BCL2_up<-gsub("1", "Elevated BCL2, n=7", data1$BCL2_up)
data1$BCL2_up<-gsub("0", "Non-elevated BCL2, n=238", data1$BCL2_up)

data1$BCL2_up<-as.factor(data1$BCL2_up)
data2<-data1
p <- ggplot(data2, aes(x=data2$BCL2_up, y=data2$X24342497, colour= data2$BCL2_up, fill=data2$BCL2_up)) + theme_bw() +
  geom_boxplot(alpha = 0.2) +
  geom_jitter(alpha = 0.75, width = 0.2) +
  guides(colour = F, fill = F) +
  theme(panel.grid = element_blank()) + labs(title="",x="", y = "Venetoclax Emax") + theme(text = element_text(size = 18))

p
```
